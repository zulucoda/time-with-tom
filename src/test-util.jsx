import React from 'react';
import { oneOfType, shape, element, func } from 'prop-types';
import { Form, Formik } from 'formik';
import { BrowserRouter as Router } from 'react-router-dom';

/**
 * Formik Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {Object} props
 * @param {Object} initialValues
 * @param {Function} onSubmit
 * @returns {Formik}
 */
export const FormikWrapperTestUtil = ({
  Component,
  props = {},
  initialValues = {},
  onSubmit = jest.fn(),
}) => (
  <Formik initialValues={initialValues} onSubmit={onSubmit}>
    <Form>
      <Component {...props} />
    </Form>
  </Formik>
);

FormikWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
  initialValues: shape({}),
  onSubmit: func,
};

/**
 * React Router Wrapper Test Util
 * @param {React.FunctionComponent} Component
 * @param {object} props
 * @returns {React.FunctionComponent}
 */
export const ReactRouterWrapperTestUtil = ({ Component, props = {} }) => (
  <Router>
    <Component {...props} />
  </Router>
);

ReactRouterWrapperTestUtil.propTypes = {
  Component: oneOfType([element.isRequired, func]),
  props: shape({}),
};
