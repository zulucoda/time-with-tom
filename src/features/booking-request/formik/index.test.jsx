import { render } from '@testing-library/react';
import React from 'react';
import BookingRequestFormik from '.';
import { ReactRouterWrapperTestUtil } from '../../../test-util';

describe('Features - Booking Request - Form - Unit Test', function () {
  test('does it render', () => {
    const { container } = render(
      <ReactRouterWrapperTestUtil>
        <BookingRequestFormik />
      </ReactRouterWrapperTestUtil>,
    );
    expect(container).toMatchSnapshot();
  });
});
