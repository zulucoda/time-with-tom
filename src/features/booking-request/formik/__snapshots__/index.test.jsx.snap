// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features - Booking Request - Form - Unit Test does it render 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="sc-Axmtr jPGtFr"
    >
      <div
        class="sc-AxirZ hoPrKv"
      >
        <div
          class="sc-AxjAm bLetVr"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
          >
            activity
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-component-select-activity"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="mui-component-select-activity"
              role="button"
              tabindex="0"
            >
              Please select
            </div>
            <input
              name="activity"
              type="hidden"
              value="none"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-AxiKw jMQjew"
        >
          <h3
            class="sc-AxhUy jNsStv"
          >
            Amount:
          </h3>
          <div
            class="sc-AxhCb cGzwOU"
          >
            R
             
            -1
          </div>
        </div>
      </div>
      <div
        class="sc-AxheI fMcmrM"
      >
        <div
          class="sc-AxgMl eTFDCq"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              name
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxgMl eTFDCq"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              surname
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="surname"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxgMl eTFDCq"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              email
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="email"
                type="email"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-fzpans bbUFkx"
      >
        <div
          class="sc-fzoLsD bhafJB"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
            format="HH:mm"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              startTime
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="startTime"
                placeholder="18:00"
                readonly=""
                type="text"
                value="02:00"
              />
            </div>
            
          </div>
        </div>
        <div
          class="sc-fzoLsD bhafJB"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
            format="HH:mm"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              endTime
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="endTime"
                placeholder="18:00"
                readonly=""
                type="text"
                value="02:00"
              />
            </div>
            
          </div>
        </div>
        <div
          class="sc-fzozJi iwRtnX"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
            format="yyyy/MM/dd"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink="true"
            >
              date
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="date"
                placeholder="2020/04/21"
                readonly=""
                type="text"
                value="2020/04/21"
              />
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="sc-fzplWN jETONs"
      >
        <div
          class="sc-AxgMl eTFDCq"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Cardholder Name
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="ccName"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxgMl eTFDCq"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Card Number
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="ccNumber"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="sc-fznyAO fWjhrI"
        >
          <div
            class="sc-AxgMl eTFDCq"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                Expiry Month
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="ccExpiryMonth"
                  type="number"
                  value="0"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-AxgMl eTFDCq"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                Expiry Year
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="ccExpiryYear"
                  type="number"
                  value="0"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-AxgMl eTFDCq"
          >
            <div
              class="MuiFormControl-root MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                CVC
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="ccCvc"
                  type="number"
                  value="0"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="sc-AxmLO sLNjy"
        type="submit"
      >
         Submit 
      </button>
    </div>
  </form>
</div>
`;
